plugins/woocommerce/src/Admin/Features/ProductBlockEditor/ProductTemplates/ProductBlock.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Automattic\WooCommerce\Admin\Features\ProductBlockEditor\ProductTemplates;
+
+use Automattic\WooCommerce\Admin\BlockTemplates\BlockInterface;
+use Automattic\WooCommerce\Admin\BlockTemplates\ContainerInterface;
+use Automattic\WooCommerce\Internal\Admin\BlockTemplates\Block;
+
+class ProductBlock extends Block implements ContainerInterface {
```

Comment:
Any reason to extend from `Block` instead of `AbstractBlock`?  My preference is the latter since `Block` is meant to be a generic block and we're implementing an `add_block` function in here already.

Reply:
Ah, actually `BlockContainerTrait` should house this and it should be removed from `Block` altogether.

Reply:
I will move it over :)